apps/cli/src/adapters/yarn/index.ts

```diff
@@ -0,0 +1,16 @@
+import { execa } from "execa";
+import * as process from "node:process";
+
+import { NodeReader, ScriptSchema } from "../../domain/node.js";
+
+export const makeNodeReader = (): NodeReader => ({
+  getScripts: async (cwd = process.cwd()) => {
+    const { exitCode, stdout } = await execa("yarn", ["run", "--json"], {
+      cwd,
+    });
+
+    return exitCode === 1
+      ? []
+      : stdout.split("\n").map((line) => ScriptSchema.parse(JSON.parse(line)));
```

Comment:
`JSON.parse` can throw (maybe we need to check it or to handle the error differently?)